spec/unit/facter/udev_spec.rb

```diff
@@ -0,0 +1,163 @@
+# frozen_string_literal: true
+
+require 'spec_helper'
+
+describe 'udev fact', type: :fact do
+  subject(:fact) { Facter.fact(:udev) }
+
+  before { Facter.clear }
```

Comment:
Some day we'll have a specific fact tester in rspec-puppet that does all of this.